anndata/_io/specs/registry.py

```diff
@@ -278,7 +278,7 @@ def write_elem(
 
         # Normalize k to abosulte path
         if not PurePath(k).is_absolute():
-            k = str(PurePath(store.name) / k)
+            k = (PurePath(store.name) / k).as_posix()
```

Comment:
This should probably just be `PurePosixPath`

Reply:
Thanks, this is indeed more elegant.